bot/cogs/schedule.py

```diff
@@ -0,0 +1,305 @@
+import shlex
+from typing import TYPE_CHECKING, List
+
+import aiohttp
+import discord
+from cogs.custom_ui.page_turn_embed import PageTurnView
+from discord.ext import commands
+
+if TYPE_CHECKING:
+    from peterbot import PeterBot
+
+
+class Schedule(commands.Cog):
+    """Handler for associated UCI Schedule commands
+
+    Parameters
+    ----------
+    bot : PeterBot
+        The bot object
+
+    Commands
+    ----------
+    soc : Searches PeterPortal API for SOC
+    """
+
+    def __init__(self, bot: "PeterBot"):
+        self.bot = bot
+
+    @commands.command(
+        name="soc",
+        brief="Search UCI SOC",
+        usage="soc [term] [flags]",
+        example="soc 2020 Fall --ge GE-1A",
+        help="Search for classes on the UCI SOC",
+        description="""
+        Flags:
+        --ge:               [ANY, GE-1A, GE-3,...]*
+        --department:       Dept. names. Ex: I&C SCI, PSYC*
+        --courseNumber      [32A, 31-33,]*
+        --division          [ALL, LowerDiv, UpperDiv, Graduate]
+        --sectionCodes      WebReg codes. Ex: 44201*
+        --instructorName    Instructor last name. Ex: Holton*
+        --courseTitle       Course name. Ex: Intro to Chem
+        --sectionType       [ALL, LEC, LAB, SEM, ...]
+        --units             Integer OR 'VAR' for variable
+        --days              [M, T, W, Th, F, MWF, ...]
+        --startTime         12hr formatted time. Ex: 1:00PM
+        --endTime           12hr formatted time. Ex: 2:00PM
+        --maxCapacity       Integer, < and > optional. Ex: <300
+        --fullCourses       [ANY, SkipFullWaitlist, FullOnly, OverEnrolled]
+        --cancelledCourses  [Exclude, Include, Only]
+        --building          Building codes. Ex: EH
+        --room              Room number. Ex: 1200
+
+        * = one of these required
+        """,
+    )
+    async def soc(self, ctx: "commands.Context", *args):
+        """Command to search PeterPortal API for SOC"""
+
+        # Parse arg string into term string and flag dict
+        # Ex:
+        #       "2022 Spring --ge GE-4 --department CHEM"
+        #   becomes
+        #       term = "2022 Spring"
+        #       flags = {"ge":"GE-4", "department":"CHEM"}
+        term = " ".join(args).split("--")[0]
```

Comment:
If a user provides the command
```
$soc --department COMPSCI
```
The department flag is parsed as the term and the command fails with a misleading error message.